### PR TITLE
Allow NoneType columns in Redshift COPY job

### PR DESCRIPTION
**Pull request by dmohns**
## Description
Preceding booleanness check lowers requirements for the columns attribute and ensures backward-compatibility.

## Motivation and Context
It fixes issue #2298.

### CODE REVIEW

**Comment by dmohns** [comment]
@dlstadther I added some basic tests for the cases you suggested. I am not too familiar with writing tests. Happy to receive further comments.

File: test/contrib/redshift_test.py [hunk 1]
```diff
@@ -214,6 +216,123 @@ def test_s3_copy_to_temp_table(self, mock_redshift_target, mock_copy):
             (task.table,),
         )
 
+    @mock.patch("luigi.contrib.redshift.RedshiftTarget")
+    def test_s3_copy_with_valid_columns(self, mock_redshift_target):
+        task = DummyS3CopyToTableKey(columns=(
+            ('some_text', 'varchar(255)'),
+            ('some_int', 'int'),
```
Comment (dlstadther) [inline test/contrib/redshift_test.py]: Since this is what is defined as the default, you should be able to just `task = DummyS3CopyToTableKey()`. Right?

Comment (dmohns) [reply test/contrib/redshift_test.py]: The idea was to emphasize on the valid columns types. But re-thinking it: I agree that it should be clear enough what the test is doing even when simply using the default `columns` attribute from `DummyS3CopyToTableKey()`.

Side note: For changes like this is there any preference whether to amend preceding commits or add them to a new commit?

Comment (dlstadther) [reply test/contrib/redshift_test.py]: A new commit is fine. We squash/merge PRs and amend the message where necessary. So no need to rebase on your end (unless you just want to).

**Comment by dlstadther** [comment]
Sorry for the delay here @dmohns . Thanks for your contribution!

**Pull request merged**